coderd/wsbuilder/wsbuilder.go

```diff
@@ -442,6 +443,20 @@ func (b *Builder) buildTx(authFunc func(action policy.Action, object rbac.Object
 
 	var workspaceBuild database.WorkspaceBuild
 	err = b.store.InTx(func(store database.Store) error {
+		names, values, err := b.getParameters()
+		if err != nil {
+			// getParameters already wraps errors in BuildError
+			return err
+		}
+
+		if b.templateVersionPresetID == uuid.Nil {
```

Comment:
This is not strictly necessary for when a prebuilt workspace is claimed, but it ensures that a workspace build always has a preset ID set if one matches.